metricshub-engine/src/test/java/org/sentrysoftware/metricshub/engine/connector/deserializer/ConnectorDeserializerTest.java

```diff
@@ -16,13 +19,33 @@ public String getResourcePath() {
 	}
 
 	@Test
-	void testDeserializeConnector() throws IOException {
+	void testDeserializeConnectorWithStringPlatformsField() throws IOException {
 		final Connector connector = getConnector("connector");
 		final ConnectorIdentity connectorIdentity = connector.getConnectorIdentity();
 
 		assertNotNull(connector);
 		assertEquals("testConnector", connectorIdentity.getDisplayName());
-		assertEquals("testPlatforms", connectorIdentity.getPlatforms());
+		final Set<String> expectedPlatforms = new LinkedHashSet<>();
+		expectedPlatforms.add("testPlatform1");
+		expectedPlatforms.add("testPlatform2");
+		expectedPlatforms.add("testPlatform3");
+		assertTrue(expectedPlatforms.containsAll(connectorIdentity.getPlatforms()));
```

Comment:
Check that the returned type is LinkedHashSet and the two LinkedHashSet are exactly the same using something like: 
```java
assertEquals(new ArrayList<>(expectedPlatforms), new ArrayList<>(actualPlatforms), "The order of platforms does not match!");
```